RULES.md

```diff
@@ -147,6 +147,35 @@ it can be made safer by converting it into:
   #                   | Replace the expression with the environment variable
 ```
 
+## <a id="ADES105"></a> ADES105 - Expression in `run:` directive with PowerShell
+
+When an expression appears in a `run:` directive you can avoid potential attacks by extracting the
+expression into an environment variable and using the environment variable instead.
+
+For example, given the workflow snippet:
+
+```yaml
+- name: Example step
+  shell: pwsh
+  run: |
+    echo 'Hello ${{ inputs.name }}'
+```
+
+it can be made safer by converting it into:
+
+```yaml
+- name: Example step
+  shell: pwsh
+  env:
+    NAME: ${{ inputs.name }} # <- Assign the expression to an environment variable
+  run: |
+    echo "Hello $Env:NAME"
+#        ^      ^^^^^^^^^
+#        |      | Replace the expression with the environment variable
+#        |
+#        | Note: the use of double quotes is required in this example (for interpolation)
```

Comment:
Confirmed in [run 12965557510](https://github.com/ericcornelissen/ades/actions/runs/12965557510)